midend/predication.cpp

```diff
@@ -43,59 +245,31 @@ const IR::Node* Predication::preorder(IR::IfStatement* statement) {
 
     ++ifNestingLevel;
     auto rv = new IR::BlockStatement;
-    cstring conditionName = generator->newName("cond");
-    auto condDecl = new IR::Declaration_Variable(conditionName, IR::Type::Boolean::get());
-    rv->push_back(condDecl);
-    auto condition = new IR::PathExpression(IR::ID(conditionName));
-
-    // A vector for a new BlockStatement.
-    auto block = new IR::BlockStatement;
-
-    const IR::Expression* previousPredicate = predicate();  // This may be nullptr
-    // a new name for the new predicate
-    cstring newPredName = generator->newName("pred");
-    predicateName.push_back(newPredName);
-    auto decl = new IR::Declaration_Variable(newPredName, IR::Type::Boolean::get());
-    block->push_back(decl);
-    // This evaluates the if condition.
-    // We are careful not to evaluate any conditional more times
-    // than in the original program, since the evaluation may have side-effects.
-    auto trueCond = new IR::AssignmentStatement(clone(condition), statement->condition);
-    block->push_back(trueCond);
-
-    const IR::Expression* pred;
-    if (previousPredicate == nullptr) {
-        pred = clone(condition);
-    } else {
-        pred = new IR::LAnd(previousPredicate, clone(condition));
-    }
-    auto truePred = new IR::AssignmentStatement(predicate(), pred);
-    block->push_back(truePred);
-
+    currentBlock = rv;
+    nestedConditions.push_back(statement->condition);
```

Comment:
This process of pushing and popping the IfStatement conditions is completely redundant with the information already tracked in the Visitor::Context -- if you use that, then these complex tracking methods can go away completely